### PR TITLE
feat(handlers): add editWebhook*(), deleteWebhook*(), getWebhook*()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no
  code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed,
    parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation,
  README, etc.

### CODE REVIEW

File: src/api/handlers/webhook.ts [hunk 1]
```diff
@@ -57,6 +58,86 @@ export async function createWebhook(
...
+  const result = await RequestManager.delete(
+    endpoints.WEBHOOK(webhookID, webhookToken),
+  );
+
+  return result;
+}
+
+/** Returns the new webhook object for the given id. */
+export function getWebhook(webhookID: string) {
+  return RequestManager.get(endpoints.WEBHOOK_ID(webhookID));
```
Comment (Skillz4Killz) [inline src/api/handlers/webhook.ts:130]: return type?

File: src/api/handlers/webhook.ts [hunk 2]
```diff
@@ -57,6 +58,86 @@ export async function createWebhook(
...
+    ...options,
+    channel_id: options.channelID,
+  });
+
+  return result;
+}
+
+/** Edit a webhook. Returns the updated webhook object on success. */
+export async function editWebhookWithToken(
+  webhookID: string,
```
Comment (Skillz4Killz) [inline src/api/handlers/webhook.ts:87]: return type

File: src/api/handlers/webhook.ts [hunk 3]
```diff
@@ -57,6 +58,86 @@ export async function createWebhook(
...
+  );
+  if (
+    !hasManageWebhooksPerm
+  ) {
+    throw new Error(Errors.MISSING_MANAGE_WEBHOOKS);
+  }
+
+  const result = await RequestManager.delete(endpoints.WEBHOOK_ID(webhookID));
+
+  return result;
```
Comment (Skillz4Killz) [inline src/api/handlers/webhook.ts:113]: return type?

File: src/api/handlers/webhook.ts [hunk 4]
```diff
@@ -57,6 +58,86 @@ export async function createWebhook(
...
+/** Delete a webhook permanently. Returns a undefined on success */
+export async function deleteWebhookWithToken(
+  webhookID: string,
+  webhookToken: string,
+) {
+  const result = await RequestManager.delete(
+    endpoints.WEBHOOK(webhookID, webhookToken),
+  );
+
+  return result;
```
Comment (Skillz4Killz) [inline src/api/handlers/webhook.ts:125]: return type?

Comment (itohatweb) [reply src/api/handlers/webhook.ts:125]: return type is undefined.

File: src/api/handlers/webhook.ts (hunk 3 above)
Comment (itohatweb) [reply src/api/handlers/webhook.ts:113]: is undefineed

File: src/api/handlers/webhook.ts [hunk 5]
```diff
@@ -57,6 +58,87 @@ export async function createWebhook(
...
+  );
+  if (
+    !hasManageWebhooksPerm
+  ) {
+    throw new Error(Errors.MISSING_MANAGE_WEBHOOKS);
+  }
+
+  const result = await RequestManager.delete(endpoints.WEBHOOK_ID(webhookID));
+
+  return result as undefined;
```
Comment (ayntee) [inline src/api/handlers/webhook.ts:113]:
```suggestion
  return result;
```

File: src/api/handlers/webhook.ts [hunk 6]
```diff
@@ -57,6 +58,87 @@ export async function createWebhook(
...
+/** Delete a webhook permanently. Returns a undefined on success */
+export async function deleteWebhookWithToken(
+  webhookID: string,
+  webhookToken: string,
+) {
+  const result = await RequestManager.delete(
+    endpoints.WEBHOOK(webhookID, webhookToken),
+  );
+
+  return result as undefined;
```
Comment (ayntee) [inline src/api/handlers/webhook.ts:125]:
```suggestion
  return result;
```

**Review by ayntee — APPROVED** [review]
LGTM

**Pull request merged**